### PR TITLE
test: add more test for pkg network

### PR DESCRIPTION
**Pull request by a631807682**
#### What type of PR is this?

test

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
添加`pkg/network`单测

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: Supplement unit test for `pkg/network`, coverage increased from 68.14% to 95.2%.
I see that  [netpoll's DialTimeout](https://github.com/cloudwego/hertz/blob/develop/pkg/network/netpoll/dial.go#L48) doesn't check tlsConfig, I'm not sure if it's a bug

https://github.com/cloudwego/hertz/blob/develop/pkg/network/netpoll/dial.go#L48

zh(optional): 补充`pkg/network`的单元测试，覆盖率从68.14%提高到了95.2%.。
我看到[netpoll的DialTimeout](https://github.com/cloudwego/hertz/blob/develop/pkg/network/netpoll/dial.go#L48)并没有检查`tlsConfig`，我不确定是不是个错误。

#### Which issue(s) this PR fixes:

related to https://github.com/cloudwego/hertz/issues/257

### CODE REVIEW

**Comment by Duslia** [comment]
Please fix CI

**Comment by Duslia** [comment]
netpoll doesn't support tls, so there is no need to care  aboud tls.config in netpoll.

**Comment by a631807682** [comment]
> netpoll does not support tls, so there is no need to care about tls.config in netpoll.

Yes, I'm not sure if we need to return `errNotSupportTLS` since it's a public package.

**Comment by Duslia** [comment]
> > netpoll does not support tls, so there is no need to care about tls.config in netpoll.
> 
> Yes, I'm not sure if we need to return `errNotSupportTLS` since it's a public package.

I think it's ok to return this.

**Comment by welkeyever** [comment]
Thanks for your report!
It would be better if the alignment returns an error - although this code will work properly - netpoll will return the "not support" error instead.

Optimize PR is welcome~

**Comment by a631807682** [comment]
> Thanks for your report! It is better to return errors to be able to align - although this code will work properly - netpoll will return the "not support" error instead.
> 
> Optimize PR is welcome~

Can I create an optimized PR after this PR is complete? because their tests may conflict. https://github.com/cloudwego/hertz/pull/672/files#diff-da9b81109f65e95b79b7f99cde6f9b44a23119bfe16028b3747b161b2e307bacR62

File: pkg/network/netpoll/dial_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 CloudWeGo Authors
```
Comment (welkeyever) [inline pkg/network/netpoll/dial_test.go:1]: AA

File: pkg/network/standard/dial_test.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
```
Comment (welkeyever) [inline pkg/network/standard/dial_test.go:2]: AA

**Comment by welkeyever** [comment]
> > Thanks for your report! It is better to return errors to be able to align - although this code will work properly - netpoll will return the "not support" error instead.
> > Optimize PR is welcome~
> 
> Can I create an optimized PR after this PR is complete? because their tests may conflict. https://github.com/cloudwego/hertz/pull/672/files#diff-da9b81109f65e95b79b7f99cde6f9b44a23119bfe16028b3747b161b2e307bacR62

Sure~

File: pkg/network/netpoll/dial_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	"github.com/cloudwego/hertz/pkg/common/config"
+	"github.com/cloudwego/hertz/pkg/common/test/assert"
+	"github.com/cloudwego/hertz/pkg/common/test/mock"
+)
+
+func TestDial(t *testing.T) {
+	t.Run("NetpollDial", func(t *testing.T) {
+		const nw = "tcp"
+		const addr = ":10100"
```
Comment (welkeyever) [inline pkg/network/netpoll/dial_test.go:35]: localhost will be better

File: pkg/network/netpoll/transport_test.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+	"github.com/cloudwego/hertz/pkg/common/config"
+	"github.com/cloudwego/hertz/pkg/common/test/assert"
+	"github.com/cloudwego/hertz/pkg/network"
+	"golang.org/x/sys/unix"
+)
+
+func TestTransport(t *testing.T) {
+	const nw = "tcp"
+	const addr = ":10103"
```
Comment (welkeyever) [inline pkg/network/netpoll/transport_test.go:37]: AA

File: pkg/network/standard/dial_test.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+	"testing"
+	"time"
+
+	"github.com/cloudwego/hertz/pkg/common/config"
+	"github.com/cloudwego/hertz/pkg/common/test/assert"
+)
+
+func TestDial(t *testing.T) {
+	const nw = "tcp"
+	const addr = ":10104"
```
Comment (welkeyever) [inline pkg/network/standard/dial_test.go:38]: addr = ":

Comment (welkeyever) [inline pkg/network/standard/dial_test.go:38]: AA

File: pkg/network/standard/dial_test.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+	assert.Nil(t, err)
+
+	nConn, err := dial.DialTimeout(nw, addr, time.Second, nil)
+	assert.Nil(t, err)
+	defer nConn.Close()
+}
+
+func TestDialTLS(t *testing.T) {
+	const nw = "tcp"
+	const addr = ":10104"
```
Comment (welkeyever) [inline pkg/network/standard/dial_test.go:61]: AA

**Review by welkeyever — APPROVED** [review]
LGTM, thx~

**Review by Duslia — APPROVED** [review]

**Pull request merged**